node/test/Test/Pos/Client/Txp/UtilSpec.hs

```diff
@@ -0,0 +1,246 @@
+    prop createMTxWorksWhenWeAreRichDesc createMTxWorksWhenWeAreRichSpec
+    prop stabilizationDoesNotFailDesc stabilizationDoesNotFailSpec
+    prop feeIsNonzeroDesc feeIsNonzeroSpec
+    prop manyUtxoTo1Desc manyUtxoTo1Spec
+    prop manyAddressesTo1Desc manyAddressesTo1Spec
+    prop manyAddressesToManyDesc manyAddressesToManySpec
+    prop redemptionDesc redemptionSpec
+    prop txWithRedeemOutputFailsDesc txWithRedeemOutputFailsSpec
+  where
+    createMTxWorksWhenWeAreRichDesc =
+        "Transaction is created successfully when we have 1 input with 1M coins " <>
+        "and 1 output with 1 coin"
+    stabilizationDoesNotFailDesc =
+        "The message \"Couldn't stabilize tx fee after 5 attempts\" is not thrown " <>
+        "when there is 1 input with 200k coins and 1 output with 1 coin"
+    feeIsNonzeroDesc =
+        "An attempt to create a tx for 1 coin when we have 100k coins fails " <>
+        "because of the fee"
+    manyUtxoTo1Desc =
+        "Transaction is created successfully when we have 10 items in Utxo " <>
+        "for a single address with 100k coins each"
+    manyAddressesTo1Desc =
+        "Transaction is created successfully when we have 10 items in Utxo " <>
+        "for 10 different addresses with 100k coins each and 1 output with 1 coin"
+    manyAddressesToManyDesc =
+        "Transaction is created successfully when we have 10 items in Utxo " <>
+        "for 10 different addresses with 100k coins each and 10 outputs with 1 coin each"
+    redemptionDesc =
+        "Redemption transaction is created successfully"
+    txWithRedeemOutputFailsDesc =
+        "An attempt to create a tx with a redeem address as an output fails"
+
+createMTxWorksWhenWeAreRichSpec :: HasCoreConstants => TxpTestProperty ()
+createMTxWorksWhenWeAreRichSpec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left err -> stopProperty $ sformat ("Failed to create tx: "%build) err
+        Right tx -> ensureTxMakesSense tx cmpUtxo cmpOutputs
+  where
+    CreateMTxParams {..} = makeManyAddressesToManyParams 1 1000000 1 1
+
+stabilizationDoesNotFailSpec :: HasCoreConstants => TxpTestProperty ()
+stabilizationDoesNotFailSpec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left err@(FailedToStabilize _) -> stopProperty $ pretty err
+        Left _                         -> return ()
+        Right tx                       -> ensureTxMakesSense tx cmpUtxo cmpOutputs
+  where
+    CreateMTxParams {..} = makeManyAddressesToManyParams 1 200000 1 1
+
+feeIsNonzeroSpec :: HasCoreConstants => TxpTestProperty ()
+feeIsNonzeroSpec = do
+    txOrError <- createMTx cmpUtxo cmpSigners cmpOutputs cmpAddrData
+    case txOrError of
+        Left (NotEnoughMoney _) -> return ()
+        Left err -> stopProperty $ pretty err
+        Right _ -> stopProperty $
+            sformat ("Transaction was created even though there were "%
```

Comment:
Why do you use sformat without any formatters inside? :confused:

Reply:
Fixed.